single-line-diagram/single-line-diagram-core/src/main/java/com/powsybl/sld/layout/AbstractZoneLayout.java

```diff
@@ -7,27 +7,36 @@
  */
 package com.powsybl.sld.layout;
 
-import com.powsybl.sld.model.graphs.SubstationGraph;
-import com.powsybl.sld.model.graphs.VoltageLevelGraph;
-import com.powsybl.sld.model.graphs.ZoneGraph;
+import com.powsybl.sld.layout.pathfinding.*;
+import com.powsybl.sld.model.graphs.*;
+import com.powsybl.sld.model.nodes.*;
 
-import java.util.Objects;
+import java.util.*;
 
 /**
  * @author Thomas Adam {@literal <tadam at silicom.fr>}
  */
 public abstract class AbstractZoneLayout extends AbstractBaseLayout<ZoneGraph> {
     protected SubstationLayoutFactory sLayoutFactory;
     protected VoltageLevelLayoutFactory vLayoutFactory;
+    protected Map<BaseGraph, AbstractLayout<SubstationGraph>> layoutBySubstation;
+    protected PathFinder pathFinder;
 
     protected AbstractZoneLayout(ZoneGraph graph, SubstationLayoutFactory sLayoutFactory, VoltageLevelLayoutFactory vLayoutFactory) {
         super(graph);
         this.sLayoutFactory = Objects.requireNonNull(sLayoutFactory);
         this.vLayoutFactory = Objects.requireNonNull(vLayoutFactory);
+        this.layoutBySubstation = new HashMap<>();
+        for (SubstationGraph subGraph : getGraph().getSubstations()) {
+            Layout sLayout = sLayoutFactory.create(subGraph, vLayoutFactory);
+            layoutBySubstation.put(subGraph, (AbstractLayout<SubstationGraph>) sLayout);
+        }
     }
 
     @Override
     public void run(LayoutParameters layoutParameters) {
+        pathFinder = new PathFinderFactory().create(layoutParameters.getZoneLayoutPathFinder());
```

Comment:
As discussed, it is not a good idea to put a new `ZoneLayoutPathFinderFactory` field in LayoutParameters (because of serialization/deserialization). Let's try something like this instead:
`ZoneLayoutFactory zoneLayoutFactory = new MatrixZoneLayoutFactory(pathFinder)`